process_linux.go

```diff
@@ -236,5 +236,14 @@ func (p *initProcess) createNetworkInterfaces() error {
 }
 
 func (p *initProcess) signal(s os.Signal) error {
-	return p.cmd.Process.Signal(s)
+	return maybeSyscall(p.cmd.Process, s)
+}
+
+func maybeSyscall(process *os.Process, s os.Signal) error {
+	if sys, ok := s.(syscall.Signal); ok {
+		// Prefer syscall.Kill because os.Process.Signal masks ESRCH
+		return syscall.Kill(process.Pid, sys)
+	} else {
+		return process.Signal(s)
+	}
```

Comment:
I think using syscall.Kill as-is is unsafe as process.Signal performs several useful safety check ( make sure that we dont do `kill -1` or `kill -0` or not having to perform `kill` at all if it's certain that the process is finished ). We at least have to check for PID -1/0 here.

What do you think about checking error string coming out of `process.Signal` (either here or in the caller) i.e. checking for `error: process already finished/released...` so we dont have to duplicate `process.Signal` logic ?